src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -10,69 +10,57 @@
 using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.PooledObjects;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.StackTraceExplorer
 {
+    using StackFrameNodeOrToken = EmbeddedSyntaxNodeOrToken<StackFrameKind, StackFrameNode>;
+    using StackFrameToken = EmbeddedSyntaxToken<StackFrameKind>;
+    using StackFrameTrivia = EmbeddedSyntaxTrivia<StackFrameKind>;
+
     /// <summary>
-    /// A line of text that was parsed by <see cref="StackTraceAnalyzer" />
-    /// to provide metadata bout the line. Expected to be the parsed output 
-    /// of a serialized <see cref="StackFrame"/>
+    /// A line from <see cref="StackTraceAnalyzer.Parse(string, CancellationToken)"/> that
+    /// was parsed by <see cref="StackFrameParser"/>
     /// </summary>
     internal sealed class ParsedStackFrame : ParsedFrame
     {
+        public readonly StackFrameTree Tree;
+
         public ParsedStackFrame(
             string originalText,
-            TextSpan typeSpan,
-            TextSpan methodSpan,
-            TextSpan argsSpan,
-            TextSpan fileSpan = default)
+            StackFrameTree tree)
             : base(originalText)
         {
-            Contract.ThrowIfTrue(typeSpan.IsEmpty);
-            Contract.ThrowIfTrue(methodSpan.IsEmpty);
-
-            TypeSpan = typeSpan;
-            MethodSpan = methodSpan;
-            ArgsSpan = argsSpan;
-            FileSpan = fileSpan;
+            Tree = tree;
         }
 
-        /// <summary>
-        /// The full type name parsed from the line. 
-        /// ex: [|Microsoft.CodeAnalysis.Editor.CallstackExplorer.|]Example(arg1, arg2)
-        /// </summary>
-        public TextSpan TypeSpan { get; }
+        public StackFrameCompilationUnit Root => Tree.Root;
 
-        /// <summary>
-        /// The method name span
-        /// ex: Microsoft.CodeAnalysis.Editor.CallstackExplorer.[|Example|](arg1, arg2)
-        /// </summary>
-        public TextSpan MethodSpan { get; }
+        public async Task<ISymbol?> ResolveSymbolAsync(Solution solution, CancellationToken cancellationToken)
+        {
+            // MemberAccessExpression is [Expression].[Identifier], and Identifier is the 
+            // method name.
+            var typeExpression = Root.MethodDeclaration.MemberAccessExpression.Left;
+            var fullyQualifiedTypeName = typeExpression.CreateString(Tree, skipTrivia: true);
```

Comment:
interesting... not sure why this is passed the Tree object.

Reply:
should just be ToString to make SyntaxNode.

Reply:
Tree object was passed because of nodes not having VirtualChars. Talked a bit offline, either going to move to having extensions return a `TextSpan` or using a stringbuilder to build from leaf tokens